### DISCUSSION TITLE
Is it possible to render multiple transparent objects in filament scene as transparent at the same time? How?

### DISCUSSION THREAD

**Discussion by 1939938853** [google/filament#6186] [Q&A]
Hello,

I'm trying to use filament based visualizer in Open3D. When doing multiple transparent objects rendering, I found that it seems only one of the transparent objects can be rendered as transparent at a time, depending on the view (camera) angle, no matter it is UnLitTransparency shader or LitTransparency shader.

Is it possible to render multiple transparent objects in filament scene as transparent at the same time? How?

Thank you!

Here is a gif-animation shows one  sphere and one cubic in a filament scene.  Both objects use UnLitTransparency shader and 50% transparency. 

![filament_transparent](https://user-images.githubusercontent.com/59957922/195664214-f7a15930-f089-40f5-9214-b55c1c05af33.gif)

**Comment by romainguy** [comment]
Yes Filament can render as many transparent objects as you want. And you can see in your example that it works. However when objects overlap the blending order will depend on the camera's viewpoint. You can control this by changing the blend priority of your object or, better yet, by avoiding overlaps.

**Reply by 1939938853** [reply]
So no way to make both overlap objects transparent? Like a overlap glass material objects? To see objects in a glass bottle that is in another  glass bottle ?

Instead of object-order rendering,  maybe filament also need to add  image-order (transparent) rendering? 

[https://vtk.org/Wiki/VTK/Tutorials/TranslucentGeometry](url)

![image](https://user-images.githubusercontent.com/59957922/195673461-88c23415-bd67-4a4d-93cd-1d43b7abf993.png)

**Reply by romainguy** [reply]
Order independent transparency is very expensive on mobile. If objects are fully enclosed in another, sorting will work. Otherwise, use blending priority.
